src/app/projects/components/project-settings/components/embeddings/embeddings.component.ts

```diff
@@ -0,0 +1,155 @@
+import { Component, Input, OnInit, SimpleChanges } from '@angular/core';
+import { Subscription } from 'rxjs';
+import { first } from 'rxjs/operators';
+import { Project } from 'src/app/base/entities/project';
+import { ProjectApolloService } from 'src/app/base/services/project/project-apollo.service';
+import { Attribute } from '../../entities/attribute.type';
+import { DownloadedModel } from '../../entities/downloaded-model.type';
+import { Embedding } from '../../entities/embedding.type';
+import { DataHandlerHelper } from '../../helper/data-handler-helper';
+import { SettingModals } from '../../helper/modal-helper';
+
+@Component({
+  selector: 'kern-embeddings',
+  templateUrl: './embeddings.component.html',
+  styleUrls: ['./embeddings.component.scss'],
+})
+export class EmbeddingsComponent implements OnInit {
+
+  @Input() project: Project;
+  @Input() attributesArrayTextUsableUploaded: Attribute[];
+  @Input() settingModals: SettingModals;
+  @Input() isManaged: boolean;
+  @Input() embeddings: Embedding[];
+  @Input() embeddingHandlesMap: { [key: string]: any };
+  @Input() dataHandlerHelper: DataHandlerHelper;
+  @Input() attributes: Attribute[];
+
+  downloadedModels: DownloadedModel[];
+  subscriptions$: Subscription[] = [];
+  somethingLoading: boolean = false;
+
+  constructor(private projectApolloService: ProjectApolloService) { }
+
+  ngOnInit(): void {
+    this.prepareDownloadedModels();
+  }
+
+  ngOnDestroy() {
+    this.subscriptions$.forEach((subscription) => subscription.unsubscribe());
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    this.checkStillLoading();
+    if (changes.embeddingHandlesMap) {
+      this.checkModelDownloaded();
+    }
+  }
+
+  prepareDownloadedModels() {
+    let downloadedModelsQuery$, downloadedModelsList$;
+    [downloadedModelsQuery$, downloadedModelsList$] = this.projectApolloService.getModelProviderInfo();
+    this.subscriptions$.push(
+      downloadedModelsList$.subscribe((downloadedModels) => {
+        this.downloadedModels = downloadedModels;
+        this.checkModelDownloaded();
+      }));
```

Comment:
im not sure that this will work as intended. The subscription is created but the query is never refetched e.g. on a websocket notification (`msgParts[1] === 'model_provider_download' && msgParts[2] === 'finished'` )

since this is never refetched it doesn't need to be pushed to notifications. However I think we should use websockets here